flake.nix

```diff
@@ -44,15 +44,20 @@
                 })) (readDir dir)
           );
         callPackage = pkgs.newScope (
-          result // {inherit callPackage nixosTests;}
+          allPackages // {inherit callPackage nixosTests;}
         );
-        args = {
+        pkgsByName = import ./pkgs/by-name {
           inherit (pkgs) lib;
           inherit callPackage;
         };
-        result = (import ./pkgs/by-name args) // (import ./pkgs args);
+        explicitPkgs = import ./pkgs {
+          inherit (pkgs) lib;
+          inherit callPackage;
+          libsForQt5 = pkgs.lib.makeScope pkgs.libsForQt5.newScope (final: allPackages);
```

Comment:
What's the motivation for adding this scope? Just because it was used in the original arpa2 repository?

Reply:
This seemed straightforward to us. Also, we're not certain we know of an alternative.

Reply:
The alternative is to not introduce a new scope, and put it in the same scope as all other packages so far. Does this break anything? What is a good argument for introducing the scope versus just using the scope we already have?

Even if a new scope was introduced, I'd prefer a solution that does not involve listing the scopes in `flake.nix`, but rather `pkgs/default.nix` or similar. Would that be possible?

Reply:
If we wanted to create the scope in `pkgs/default.nix` we would have to pass `pkgs` into it. We noticed that `pkgs` is currently not passed there; we're not sure why, so we arbitrarily decided to only pass the new `libsForQt5`. Would you have us pass `pkgs` into `pkgs/default.nix` and create the scope there?

Reply:
Yes.

Reply:
The reasoning behind the new scope is as follows. tlspool-gui depends on tlspool. Also, it seems that typically in nixpkgs, qt apps are called from that scope. So creating a new scope from that scope and with the addition of our flake's packages would provide that package with an idiomatic signature and the dependency it requires.

Having said that, we changed our minds and force-pushed that commit. There's no new scope now.